link/internal/services/machine.go

```diff
@@ -22,7 +22,7 @@ type createMachineParams struct {
 	IsActive         *bool  `json:"isactive,omitempty"`
 }
 
-func CreateMachine(data []byte, log *slog.Logger) ([]byte, error) {
+func CreateMachine(data []byte, log *slog.Logger, url string) ([]byte, error) {
```

Comment:
again - []byte params looks wrong a little

check twice is url required - why should it be param, not hardcoded constant.